pkg/appdefinition/lookup.go

```diff
@@ -8,6 +8,18 @@ import (
 	"k8s.io/apimachinery/pkg/api/equality"
 )
 
+var semantic = equality.Semantic.Copy()
+
+func init() {
+	// Add custom equality functions
+	z.Must(semantic.AddFunc(func(a, b v1.AcornBuild) bool {
+		return a.OriginalImage == b.OriginalImage &&
+			a.Context == b.Context &&
+			a.Acornfile == b.Acornfile &&
+			equality.Semantic.DeepEqual(a.BuildArgs.GetData(), b.BuildArgs.GetData())
```

Comment:
This was added to satisfy a preexisting requirement in the code that depends on equality between zero value of `v1.GenericMap` and `nil`; when we switched from a named map type to a nil, the default equality provided by `equality.Semantic` were no longer sufficient.